### DISCUSSION TITLE
Silent Printing

### DISCUSSION THREAD

**Discussion by mmrashad** [rawilk/laravel-printing#53] [Q&A]
From the description, I can already tell I may indeed love to use this, but my most important question is, does sending the print job directly to a printer on the network mean it's a silent print request? Or will it also pull up the browser's print setup dialog?

**Comment by rawilk** [comment]
The server is sending the request directly to print node, so a print dialog will not be opened in the browser since it is not needed.

**Comment by lmezach** [comment]
Greetings, I need the nodeprint api key to be dynamic I only get the following error : You must provide an api key for the PrintNode driver

use Rawilk\Printing\Facades\Printing;

       // Set the API key
        Printing::driver('printnode')->setApiKey('tu-api-key');

        // Printer ID
        $printerId = 'id_impresora';

        // Path to the PDF file you want to print
        $filePath = 'ruta_al_archivo.pdf';

        // Create a new print job and send it to the printer
        $printJob = Printing::newPrintTask()
            ->printer($printerId)
            ->file($filePath)
            ->send();

       $printJob->id();

thanks, if you can help me
